### ISSUE TITLE
Convert GeointerfaceMakie into a weak dependency

### ISSUE THREAD

**Issue by felixcremer** [JuliaGeo/GeoInterface.jl#99]
Should GeointerfaceMakie be a weak dependency in Julia 1.9? This would enable the Makie recipes for all packages that implement the geointerface e.g. archgdal

**Comment by rafaqz** [comment]
Unfortunately all the package actually need to apply the macro to their own types because we Makie catch dispatch on traits.

But we could make it a weak dependency everywhere, or even just a dependency as its only MakieCore.jl. But maybe weak is best.

Someone just has to make the PRs: 

- [ ] Shapefile.jl
- [ ] ArchGDAL.jl
- [ ] LibGEOS.jl
- [ ] GeoJSON.jl

**Comment by rafaqz** [comment]
I've got a PR to update GeoInterfaceMakie.jl so vectors of geometries and Features/FeatureCollections also work. So these PRs should add the macro to geometries, features and feature collections at the same time where the packages have them.

**Comment by felixcremer** [comment]
I tried to load GeoInterfaceMakie in a weak dependency for ArchGDAL which would be loaded as soon as Makie is loaded, but I ran into world age problems. I now added GeoInterfaceMakie  as a plain dependency, because that seems to be easier.

**Comment by asinghvi17** [comment]
Could we also make it a weak dependency of GeoInterface, to get wrapper geometries enabled?  I think that should be fine, just to add a `GeoInterfaceMakieExt` to GeoInterface in addition to the regular GeoInterfaceMakie.

**Comment by rafaqz** [comment]
Wont we get some circular dependencies? Makie depends on GeoInterface.jl via GeometryBasics.jl

We should make that an extension

**Comment by asinghvi17** [comment]
I don't think so - GeoInterfaceMakie will only ever load after GeoInterface and Makie are both loaded, i.e., after Makie is loaded (because that triggers the check for applicable extensions AFAIK).

We could make GeoInterface an extension to GB easily, though.  It's just one file which doesn't define any exportable methods (we can move the definition of `GB.geointerface_geomtype` as an empty shell somewhere else in GB proper).

**Comment by rafaqz** [comment]
Ok, give it a try then

**Comment by asinghvi17** [comment]
https://github.com/JuliaGeometry/GeometryBasics.jl/pull/212

**Comment by rafaqz** [comment]
Done
